src/test/components/elements/ReportAccordionButton.test.tsx

```diff
@@ -0,0 +1,100 @@
+import { screen } from "@testing-library/react";
+import {
+  ReportAccordionButton,
+  ReportAccordionButtonProps,
+} from "../../../app/components/elements/ReportAccordionButton";
+import { renderTestEnvironment } from "../../utils";
+import * as actions from "../../../app/state/actions/logging";
+import userEvent from "@testing-library/user-event";
+
+const renderReportButton = (props: ReportAccordionButtonProps) => {
+  renderTestEnvironment({
+    role: "STUDENT",
+    PageComponent: ReportAccordionButton,
+    componentProps: props,
+    initialRouteEntries: ["/"],
+  });
+};
+
+const button = () =>
+  screen.getByRole("button", {
+    name: /report a problem/i,
+  });
+
+const testCases = [
+  { prop: "pageId", value: undefined },
+  { prop: "pageId", value: "example_id" },
+  { prop: "sectionId", value: undefined },
+  { prop: "sectionId", value: "example_id" },
+  { prop: "sectionTitle", value: undefined },
+  { prop: "sectionTitle", value: "example_title" },
+  { prop: "sectionIndex", value: undefined },
+  { prop: "sectionIndex", value: 2 },
+];
+
+describe("ReportAccordionButton", () => {
+  jest.spyOn(window, "open").mockImplementation(jest.fn());
+  const logActionSpy = jest.spyOn(actions, "logAction");
+
+  it.each(testCases)("renders button when $prop is $value", ({ prop, value }) => {
+    renderReportButton({ [prop]: value });
+    expect(button()).toBeInTheDocument();
+  });
+
+  it("opens a new window with expected URL when clicked, if pageId is provided", async () => {
+    renderReportButton({ pageId: "example_id" });
+    await userEvent.click(button());
+    expect(window.open).toHaveBeenCalledWith(
+      `/contact?preset=contentProblem&page=example_id&url=${window.location.href}`,
```

Comment:
If it's not too hard, it may be worth mocking `window.location.href`, so that you can fully control that value. For all we know, that value may be empty while the tests run.

Reply:
In this case I did initially try mocking it, but it wasn't behaving as expected. I did double check that the value during tests is localhost, so decided to leave it without mocking. Similarly I would imagine that if the test was run on staging it would be the staging URL etc - essentially just trying to confirm that whatever the location is gets handed into the URL.  I can try again to mock this if you think it would be better though

Reply:
To be honest, I don't have any suggestions on how to mock it, I would have to google it. If you've already done it and it's complicated or awkward, then that's fine, it's not the end of the world. I was asking just in case it was easy to do.

Reply:
Yeah it wasn't especially complicated, I was just doing it wrong :D added that for both the button test files.